### ISSUE TITLE
dnd list inside modal will render items under the modal when dragging

### ISSUE THREAD

**Issue by ksjogo** [uber/baseweb#3358]
## Current Behavior

dnd list inside modal will render items under the modal when dragging

## Expected Behavior

dnd list inside modal will render items over the modal when dragging

## Your Environment

| Tech    | Version |
| ------- | ------- |
| Base UI | ^9.74.0  |
| React   | ^16.13.1        |
| browser | Firefox        |

- [x] I have searched the [issues](https://github.com/uber/baseweb/issues) of this repository and believe that this is not a duplicate.

**Comment by gergelyke** [comment]
@ksjogo would you mind adding a stackblitz repro?

**Comment by nadiia** [comment]
Is it BaseWeb's modal? Do you have any `zIndex` set on the `BaseProvider`?

**Comment by ksjogo** [comment]
Yes, it is baseweb's modal and a zindex is set. It seems the transformed li is created without using the base provider?
